src/compiler/scanner.ts

```diff
@@ -429,6 +429,7 @@ namespace ts {
               case CharacterCodes.slash:
                   // starts of normal trivia
               case CharacterCodes.lessThan:
+              case CharacterCodes.bar:

```

Comment:
From further discussion with @DanielRosenwasser, it would change the public API since `SyntaxKind` would also need to change. So I think we should just live with `bar`.